### PR TITLE
Make web console alerts dismissable

### PR DESCRIPTION
**Pull request by spadgett**
Adds a close button for each alert.

openshift_web_console

See also #6187. (I think the changes here will work with that PR.)

### CODE REVIEW

**Comment by spadgett** [comment]
@jwforres PTAL
@benjaminapetersen CC

File: assets/app/views/_alerts.html [hunk 1]
```diff
@@ -1,20 +1,20 @@
...
-      <strong ng-if="alert.message" style="margin-right: 5px;">{{alert.message}}</strong><span ng-if="alert.details">{{alert.details}}</span>
-    </div>
+<div ng-repeat="(id, alert) in alerts">
+  <div class="alert" ng-class="{
+    'alert-danger': alert.type === 'error',
+    'alert-warning': alert.type === 'warning',
+    'alert-success': alert.type === 'success',
+    'alert-info': !alert.type || alert.type === 'info'
+  }">
+    <button ng-click="dismiss(id)" type="button" class="close" aria-hidden="true">
```
Comment (jwforres) [inline assets/app/views/_alerts.html]: aria-hidden true on the pficon and an sr-only span that says "Close" ?

**Comment by jwforres** [comment]
sweep the code and make sure $scope.alerts is always being set to a map, I know at one point some places snuck in where it was being set to an array instead

**Comment by jwforres** [comment]
otherwise LGTM

**Comment by spadgett** [comment]
@jwforres Good catch, TaskList uses an array.

**Comment by spadgett** [comment]
I was looking at switching to the angular-patternfly `pf-notification` directive, which supports dismiss, but unfortunately it requires us to switch from our own `alerts` to `pf-notification-list` everywhere. Not sure we want to do that.

**Comment by spadgett** [comment]
@jwforres updated the close button to work with objects or arrays and added the sr-only label

**Comment by spadgett** [comment]
[test]

**Comment by spadgett** [comment]
[test]

**Comment by spadgett** [comment]
[test]

**Comment by benjaminapetersen** [comment]
Looks like a nice/simple fix to address the main issue we have talked about with alerts.  If sufficient, shall I [close this PR](https://github.com/openshift/origin/pull/6187)?

**Comment by spadgett** [comment]
> Looks like a nice/simple fix to address the main issue we have talked about with alerts. If sufficient, shall I close this PR?

I still think it's a good refactor... Maybe for when we can bring in Patternfly toast notifications?

**Comment by jwforres** [comment]
[merge]

**Pull request merged**